docs/src/design/ten-thousand-foot-view.md

```diff
@@ -0,0 +1,211 @@
+# Ten Thousand Foot View
+
+![](./ten-thousand-foot-view-diagram.png)    
+*Figure 1: MapLibre Native Components – Ten Thousand Foot view*
+
+From ten thousand foot, MapLibre Native is composed of *Map View* and a
+*Renderer*. MapLibre Native GL supports iOS, Android, Linux, QT, MacOS,
+and nodeJS at the time of writing. Each of these *platforms* have their
+own *Map View* and *Map Renderer* component. A *Map Renderer* talks to a
+shared codebase that renders map by using device GPU. This shared piece
+of code at the time of writing supports OpenGL as its choice of
+rendering API.
+
+Apart from the platform code, *MapLibre Native GL* offers shared
+implementation for *Style, Layers, Text, Glyphs*, and *Tiles*.
+
+To summarize:
+
+1.  ***Map View*** is responsible for viewing a *slippy map,* a term
+    referring to web maps which lets a user zoom and pan around. Each
+    platform has its own *Map View*.
+
+2.  ***Map Renderer*** is composed of two parts, one is a platform
+    specific map rendering loop. The other is a cross-platform renderer
+    backend, denoted as *Renderer* in Figure 1.
+
+3.  A *map renderer* uses a ***Tile Worker*** to render individual map
+    tile. It uses an ***Actor Framework*** to send messages to a ***Tile
+    Worker*** to produce said tiles.
+
+4.  A *tile worker* prepares a tile one layer at a time. A map tile is
+    composed of multiple *layers*.
+
+5.  A ***Layer*** requires ***Style***, ***Glyphs***, and ***Sprites***
+    for to be ready for rendering.
```

Comment:
Should this also reference tile/source data? (It can't render anything until it has that.)

Generally, I'm not sure if we need to / want to discuss tile sources at this stage. Given it's whole separate system for loading and processing tiles, GeoJSON, etc. from various sources (local or remote), I'm not sure where it should be documented, but I think it probably should be.

Reply:
I think it should be documented. I did not dive straight into tiles because I talk about it later when I dive into tile workers. But it is a good concern. I will try to throw a line here or two.

Reply:
I addressed it by adding 2 more lines about Layers and Tiles.